.github/workflows/dotnet-format.yml

```diff
@@ -17,10 +17,10 @@ jobs:
       uses: actions/setup-dotnet@v4
 
     - name: dotnet restore
-      run: dotnet restore
+      run: dotnet restore OpenTelemetry.sln
 
     - name: dotnet format
-      run: dotnet format OpenTelemetry.sln --no-restore --verify-no-changes
+      run: dotnet format OpenTelemetry.sln --no-restore --verify-no-changes # Note: .proj files are currently not supported by dotnet format
```

Comment:
Ideally we would run dotnet format over `OpenTelemetry.proj` but that isn't supported currently. This means projects not added to the solution won't have this linting performed. I tried some powershell to find all the `.csproj` files and manually call dotnet format over them but I canceled my test run after 17+ minutes. IMO having it run quickly and potentially missing things is the lesser evil than covering everything and it taking forever to run.